SWIGGYDEMO_REACT/swiggy-react-app/src/components/itemDetailsAndCart/components/listByCategory/ListbyCategory.js

```diff
@@ -0,0 +1,162 @@
+            })
+        }
+
+        else {
+            category.forEach( (i) => {
+                let arrayOfItems = [];
+                arrayOfItems = data.filter( (d) => {
+                    return d.category === i && (d.name.toUpperCase().indexOf(searchInputText.toUpperCase()) > -1);
+                } )
+                if(arrayOfItems.length != 0) {
+                    arrayOfSearchedItems.push(arrayOfItems);
+                    searchCat.push(i);
+                }
+            })
+
+            category.forEach( (i) => {
+                let arrayOfItems = [];
+                arrayOfItems = data.filter( (d) => {
+                    return d.category === i;
+                } )
+                if(arrayOfItems.length != 0) {
+                    arrayOfVegOnlyItems.push(arrayOfItems);
+                    filterCat.push(i);
+                }
+            })
+        }
+        setSearchResultItems([...arrayOfSearchedItems]);
+        setItemsOrderedByCategory([...arrayOfVegOnlyItems]);
+
+        setCategoryListForSearch([...searchCat]);
+        setCategoryListForFilter([...filterCat]);
+
+        onFilter(filterCat);
+    }
+
+    useEffect( () => {
+        updateItems();
+    }, [isVeg, searchInputText])
+
+    return (
+        <>
+            {
+                searchInputText != 0 
+                ?
+                <div className='items'>
+                    <div className = "searchResult-name">Search Result</div>
+                    {
+                        searchResultItems.length != 0 ?
+                        categoryListForSearch.map( (currentCategory, index) => {
+                            return (
+                                    <div key={index} className="item">
+                                        <div className="name">
+                                            <span className="n">{currentCategory}</span><br />
+                                            <span className="i">{searchResultItems[index].length} items</span>
+                                        </div>
+                                        <div className="listItems">
+                                            <Items data={searchResultItems[index]} cartItems={cartItems} onClickADD={onClickADD} onClickPlus={onClickPlus} onClickMinus={onClickMinus}/>
+                                        </div>
+                                    </div>
+                            );
+                        })
+                        : 
+                        <div className = "searchResultEmpty">Search Result Empty</div>
+                    }
+                    <hr />
+                              
+                    {
+                        categoryListForFilter.map( (currentCategory, index) => {
+                                if (itemsOrderedByCategory[index].length !== 0) {
+                                    return (
+                                        <div id={index + 1} key={index} className="item">
```

Comment:
why do we have to add id to div?

Reply:
OnClicking to archor tag scroll to particular div with that id